plugins/zenoh-plugin-storage-manager/src/replica/digest.rs

```diff
@@ -290,50 +293,59 @@ impl Digest {
 
         // reconstruct updated parts of the digest
         for sub in subintervals_to_update {
-            // order the content, hash them
-            let checksum = Digest::get_subinterval_checksum(
-                &subintervals
-                    .get(&sub)
-                    .unwrap()
-                    .content
-                    .clone()
-                    .into_iter()
-                    .collect::<Vec<LogEntry>>(),
-            );
+            let content = &subintervals.get(&sub).unwrap().content;
```

Comment:
Are you sure this `unwrap()` cannot panic ?
Also: you could directly call `get_mut()` rather than `get()`. It would avoid the 2nd `get_mut()` below.

Reply:
For the `unwrap()`, yes, they cannot panic. This is because `subintervals_to_update` will only contain the keys that are already present in `subintervals`.
I will use `get_mut()` instead of `get()`.